### PR TITLE
Ditch commands and make new Dungeon class

### PR DESCRIPTION
**Pull request by monzo94**
Ditch commands completely.

Old Dungeon class has been renamed to Level. A Level is a single map in the game. The new Dungeon class is a collection of levels, which are interconnected via stairs going up and down.
Now the DisplayManager doesn't have to do map generation, since the dungeon takes care of that.

I've also fixed the style a bit (since you said you were cool with doing it like I was doing it at first, which imo looks more organised) and some minor bugs (one parameter not being used in DisplayManager, and updated requirements.txt since the old version of tdl wasn't installing correctly anymore).

### CODE REVIEW

File: action_manager.py [hunk 1]
```diff
@@ -77,26 +71,61 @@ def handle_key_input(cls):
...
+            return cls.pickup_action()
 
         else:
             return False
+
+    def movement_action(cls, move_direction):
+        """
+        Player tries to perform a movement action in the given direction.
+        If the tile he's moving to is unoccupied, he will move to it.
+        If it's occupied by another entity, the player will attack it instead.
```
Comment (Elkasitu) [inline action_manager.py]: Items are entities, making this statement false

File: display_manager.py [hunk 2]
```diff
@@ -48,35 +39,17 @@ class DisplayManager(metaclass=Singleton):
 
     # TODO: give consoles a better name
     tdl.set_font(get_abs_path('lucida10x10_gs_tc.png'), greyscale=True, altLayout=True)
-    console = tdl.Console(MAP_WIDTH, MAP_HEIGHT)
+    # TODO: Instead of using the level width, use views with fixed width
+    # FIXME: Since we don't have views yet, hardcode level width and height
+    console = tdl.Console(80, 40)
     panel = tdl.Console(SCREEN_WIDTH, PANEL_HEIGHT)
     backpack = tdl.Console(BACKPACK_WIDTH, SCREEN_HEIGHT)
     root_console = tdl.init(SCREEN_WIDTH, SCREEN_HEIGHT, title=GAME_TITLE,
```
Comment (Elkasitu) [inline display_manager.py]: I told you to move this into an `__init__` !! 😡

File: main.py [hunk 3]
```diff
@@ -16,17 +18,21 @@ def main():
...
+    # Initialize Dungeon
+    dungeon = Dungeon()
+    dungeon.initialize(player)
+
     # Initialize Display Manager
     display_manager = DisplayManager()
-    display_manager.gen_map(player)
-    display_manager.cur_map.entities.append(player)
+    display_manager.player = player
+    display_manager.dungeon = dungeon
```
Comment (Elkasitu) [inline main.py]: This is a big no-no, you should not inject global variables into a module, this is called monkey-patching and should only be done as a last resource with third-party libs/programs, you can pass them as arguments to the DisplayManager constructor instead.

File: main.py [hunk 4]
```diff
@@ -16,17 +18,21 @@ def main():
...
     message("Hello world!", Colors.RED)
 
     # Initialize Action Manager
     action_manager = ActionManager()
     action_manager.player = player
     action_manager.display_manager = display_manager
-    action_manager.update(display_manager.cur_map)
+    action_manager.dungeon = dungeon
```
Comment (Elkasitu) [inline main.py]: Ditto

File: misc.py [hunk 5]
```diff
@@ -101,7 +98,6 @@ class Colors:
 
 
 class RenderPriority(Enum):
-
```
Comment (Elkasitu) [inline misc.py]: Do you have an automatic formatter or something? Is this non-PEP8 compliant?

File: dungeon.py [hunk 6]
```diff
@@ -1,281 +1,94 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
```
Comment (Elkasitu) [inline dungeon.py]: ???

File: misc.py (hunk 5 above)
Comment (monzo94) [reply misc.py]: I do have an automatic formatter. Guess it follows [these](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) guidelines.

File: dungeon.py (hunk 6 above)
Comment (monzo94) [reply dungeon.py]: Whoops, forgot about it in the new file. But now that I think about it, is the shebang really necessary in files that are not *main.py*? I mean, it's the only file that should be executed anyway.

Comment (Elkasitu) [reply dungeon.py]: it doesn't hurt to have it, but the coding should always be specified

**Pull request merged**